src/Microsoft.DocAsCode.Build.RestApi/ApplyOverwriteDocumentForRestApi.cs

```diff
@@ -69,17 +69,26 @@ public IEnumerable<RestApiTagViewModel> GetTagItemsToOverwrite(FileModel article
 
         protected override void ApplyOverwrite(IHostService host, List<FileModel> overwrites, string uid, List<FileModel> articles)
         {
-            if (articles.Any(a => uid == ((RestApiRootItemViewModel)a.Content).Uid))
+            // 'articles' are filtered by registered uid, need further filtering by uid equality, then call getItemsToOverwrite function to select corresponding items
+            var matchedArticles = articles.Where(a => uid == ((RestApiRootItemViewModel)a.Content).Uid).ToList();
+            if (matchedArticles.Count > 0)
             {
-                ApplyOverwrite(host, overwrites, uid, articles, GetRootItemsFromOverwriteDocument, GetRootItemsToOverwrite);
+                ApplyOverwrite(host, overwrites, uid, matchedArticles, GetRootItemsFromOverwriteDocument, GetRootItemsToOverwrite);
+                return;
             }
-            else if (articles.Any(a => ((RestApiRootItemViewModel)a.Content).Children.Any(c => uid == c.Uid)))
+
+            matchedArticles = articles.Where(a => ((RestApiRootItemViewModel)a.Content).Children.Any(c => uid == c.Uid)).ToList();
+            if (matchedArticles.Count > 0)
             {
                 ApplyOverwrite(host, overwrites, uid, articles, GetChildItemsFromOverwriteDocument, GetChildItemsToOverwrite);
```

Comment:
...my mistake, should update.

---
In reply to: [126074254](https://github.com/dotnet/docfx/pull/1840#discussion_r126074254) [](ancestors = 126074254)